pyiceberg/io/pyarrow.py

```diff
@@ -1508,7 +1512,7 @@ def _record_batches_from_scan_tasks_and_deletes(
             if self._limit is not None and total_row_count >= self._limit:
                 break
             batches = _task_to_record_batches(
-                self._fs,
+                _fs_from_file_path(task.file.file_path, self._io),
```

Comment:
nice! I think this solves #1041 as well.

Reply:
Yep, it is :-)

Reply:
Nit: I noticed that we first pass in the path here, and the IO as a second argument. How about unifying this?